### PR TITLE
kconfig: use template for partition manager config

### PR DESCRIPTION
**Pull request by hakonfam**
Leverage the kconfig template functionality to have cleaner
Kconfig code which is also less error prone.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

### CODE REVIEW

**Comment by thst-nordic** [comment]
this error looks related.  https://jenkins-ncs.nordicsemi.no/blue/organizations/jenkins/fw-nrfconnect-nrf/detail/master/856/tests

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -14,36 +14,21 @@ config MCUBOOT
...
-	help
-	  Flash space set aside for the storage area.
-
-# Define used by partition_manager.py to deduce size of partition
-config PM_PARTITION_SIZE_MCUBOOT_PAD
-	hex "Flash space reserved for padding area."
-	default 0x200
-	help
-	  Flash space set aside for the padding area.
+partition=MCUBOOT_PAD
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig]: Prompt `Flash space reserved for padding area.` became `Flash space reserved for MCUBOOT_PAD.`

Would be nice to have a comment that MCUBOOT_PAD is the padding area.

File: boot/zephyr/Kconfig [hunk 2]
```diff
@@ -14,36 +14,21 @@ config MCUBOOT
...
 
-# Define used by partition_manager.py to deduce size of partition
-config PM_PARTITION_SIZE_MCUBOOT
-	hex "Flash space reserved for bootloader."
-	default 0xc000
-	help
-	  Flash space set aside for the MCUBoot. Note, the name
-	  of this configuration needs to match the requirements set by the
-	  script 'partition_manager.py'. See pm.yaml.
+partition=MCUBOOT
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig]: add comment: Note, the name of this configuration needs to match the requirements set by the script 'partition_manager.py'. See pm.yaml."

File: boot/zephyr/Kconfig (hunk 1 above)
Comment (hakonfam) [reply boot/zephyr/Kconfig]: The description of the MCUBOOT_PAD area is provided in the `pm.yml` which defines this partition.

File: boot/zephyr/Kconfig (hunk 2 above)
Comment (hakonfam) [reply boot/zephyr/Kconfig]: Fixed

**Pull request merged**